corehq/apps/export/models/new.py

```diff
@@ -235,42 +293,93 @@ def _generate_schema_from_xform(xform, langs, appVersion):
+        :returns: Returns a ExportDataSchema instance
         """
+        app_build_ids = get_all_app_ids(domain)
+        all_case_schema = CaseExportDataSchema()
 
-        schema = ExportDataSchema()
+        for app_doc in iter_docs(Application.get_db(), app_build_ids):
+            app = Application.wrap(app_doc)
+            case_property_mapping = get_case_properties(
+                app,
+                [case_type],
+                include_parent_properties=False
+            )
+            case_schema = CaseExportDataSchema._generate_schema_from_case_property_mapping(
+                case_property_mapping,
+                app.version,
+            )
+            case_history_schema = CaseExportDataSchema._generate_schema_for_case_history(
+                case_property_mapping,
+                app.version,
+            )
 
-        def resolvefn(group_schema1, group_schema2):
-            group_schema = ExportGroupSchema(
-                path=group_schema1.path,
-                last_occurrence=max(group_schema1.last_occurrence, group_schema2.last_occurrence),
+            all_case_schema = CaseExportDataSchema._merge_schemas(
+                all_case_schema,
+                case_schema,
+                case_history_schema
             )
-            items = _merge_lists(
-                group_schema1.items,
-                group_schema2.items,
-                keyfn=lambda item: '{}:{}'.format(_list_path_to_string(item.path), item.doc_type),
-                resolvefn=lambda item1, item2: item1.__class__.merge(item1, item2),
-                copyfn=lambda item: item.__class__(item.to_json()),
+
+        return all_case_schema
+
+    @staticmethod
+    def _generate_schema_from_case_property_mapping(case_property_mapping, appVersion):
+        # we should only generate for one case type
+        assert len(case_property_mapping.keys()) == 1
+        schema = CaseExportDataSchema()
+
+        for case_type, case_properties in case_property_mapping.iteritems():
+            group_schema = ExportGroupSchema(
+                path=[case_type],
+                last_occurrence=appVersion,
             )
-            group_schema.items = items
-            return group_schema
+            for prop in case_properties:
+                group_schema.items.append(ScalarItem(
+                    path=[prop],
+                    label=prop,
+                    last_occurrence=appVersion,
+                ))
 
-        group_schemas = _merge_lists(
-            schema1.group_schemas,
-            schema2.group_schemas,
-            keyfn=lambda group_schema: _list_path_to_string(group_schema.path),
-            resolvefn=resolvefn,
-            copyfn=lambda group_schema: ExportGroupSchema(group_schema.to_json())
+            schema.group_schemas.append(group_schema)
+
+        return schema
+
+    @staticmethod
+    def _generate_schema_for_case_history(case_property_mapping, appVersion):
```

Comment:
hey @snopoke this is for generate the group schema for the case history

<img width="1186" alt="screen shot 2016-01-27 at 10 22 52 am" src="https://cloud.githubusercontent.com/assets/918514/12617983/f5caf61c-c4df-11e5-9e7a-58b68548a6fe.png">

each group schema seems to be pretty highly customized, so i thought it made sense to just make a separate function for each one instead of try and generalize for all of them and have loads of if statements. i'll add a doc string